src/Storages/MergeTree/MergeTreeReadPool.cpp

```diff
@@ -44,13 +52,40 @@ MergeTreeReadPool::MergeTreeReadPool(
     , parts_ranges(std::move(parts_))
     , predict_block_size_bytes(context_->getSettingsRef().preferred_block_size_bytes > 0)
     , do_not_steal_tasks(do_not_steal_tasks_)
+    , merge_tree_use_const_size_tasks_for_remote_reading(context_->getSettingsRef().merge_tree_use_const_size_tasks_for_remote_reading)
     , backoff_settings{context_->getSettingsRef()}
     , backoff_state{threads_}
 {
+    if (std::ranges::count(is_part_on_remote_disk, true))
```

Comment:
hm, `is_part_on_remote_disk` is initialized in `fillPerPartInfo` below this check, so this check is always false?

Reply:
what a shame )